src/olympia/addons/tests/test_models.py

```diff
@@ -1102,60 +1102,72 @@ def test_newlines_less_than_tight(self):
 
         assert self.newlines_helper(before) == after
 
-    def test_app_numeric_slug(self):
-        cat = Category.objects.get(id=22)
-        cat.slug = 123
-        with self.assertRaises(ValidationError):
-            cat.full_clean()
-
     def test_app_categories(self):
-        def addon():
+        def get_addon():
             return Addon.objects.get(pk=3615)
 
-        c22 = Category.objects.get(id=22)
-        c22.db_name = 'CCC'
-        c22.save()
-        c23 = Category.objects.get(id=23)
-        c23.db_name = 'BBB'
-        c23.save()
-        c24 = Category.objects.get(id=24)
-        c24.db_name = 'AAA'
-        c24.save()
-
-        cats = addon().all_categories
-        assert cats == [c22, c23, c24]
-        for cat in cats:
+        # This add-on is already associated with three Firefox categories
+        # using fixtures: Bookmarks, Feeds, Social.
+        FIREFOX_EXT_CATS = CATEGORIES[amo.FIREFOX.id][amo.ADDON_EXTENSION]
+        expected_firefox_cats = [
+            FIREFOX_EXT_CATS['bookmarks'],
+            FIREFOX_EXT_CATS['feeds-news-blogging'],
+            FIREFOX_EXT_CATS['social-communication']
+        ]
+
+        addon = get_addon()
+        assert set(addon.all_categories) == set(expected_firefox_cats)
+        for cat in addon.all_categories:
             assert cat.application == amo.FIREFOX.id
 
-        cats = [c24, c23, c22]
-        app_cats = [(amo.FIREFOX, cats)]
-        assert addon().app_categories == app_cats
+        assert addon.app_categories == {amo.FIREFOX: expected_firefox_cats}
 
-        c = Category(application=amo.THUNDERBIRD.id, db_name='XXX',
-                     type=addon().type, count=1, weight=1)
-        c.save()
-        AddonCategory.objects.create(addon=addon(), category=c)
-        c24.save()  # Clear the app_categories cache.
-        app_cats += [(amo.THUNDERBIRD, [c])]
-        assert addon().app_categories == app_cats
+        thunderbird_static_cat = (
+            CATEGORIES[amo.THUNDERBIRD.id][amo.ADDON_EXTENSION]['tags'])
```

Comment:
This test is hard to follow with the setup for Thunderbird in a different style than Firefox and half-way through the method.  Can you move it further up and make it similar?